core/trino-main/src/test/java/io/trino/operator/scalar/FunctionAssertions.java

```diff
@@ -943,14 +944,16 @@ private static Split createNormalSplit()
 
     private static RowType createTestRowType(int numberOfFields)
     {
-        Iterator<Type> types = Iterables.<Type>cycle(
-                BIGINT,
-                INTEGER,
-                VARCHAR,
-                DOUBLE,
-                BOOLEAN,
-                VARBINARY,
-                RowType.from(ImmutableList.of(RowType.field("nested_nested_column", VARCHAR)))).iterator();
+        Iterator<Type> types = Stream.generate(() -> ImmutableList.<Type>of(
```

Comment:
I would suggest we keep `cycle`. Nicer to read.

Reply:
But then you have to convert it to `Stream` to do transformations?

Reply:
Yeah - only when we need any extra transformations. I can live with generate + flatmap. But it is not super readable.